### ISSUE TITLE
nest model training

### ISSUE THREAD

**Issue by wyfSunflower** [microsoft/LightGBM#3374]
Suppose there is a feature set X = {x1, x2, x3, ... xn}, and X1={xi1, xi2, xi3, ... xim} , X2 = {xj1, xj2, xj3, ..., xjn} are subset of X,
using lgb I construct two base models U = U(X1), V = V(X2)
and I want to using lgb to construct a complex model W = W(U, V, X) = W( U(X1), V(X2), X)
= W(U(xi1, xi2 ,xi3, ... xim), V(xj1, xj2, xj3, ..., xjn), x1, x2, x3, ... xn)
I don't  how to traing the total model,
 i,e. How to pass gradient from loss function to base model ,to param tuning.
Thanks!

**Comment by btrotta** [comment]
If I understand correctly, you would like to train at the same time the 2 base models U, V, and the complex model W which uses their output, and optimize the weights of all models to minimize the loss of W? This is not possible in LightGBM.

**Comment by wyfSunflower** [comment]
Another question:How to use lgb to train a gbdt+lr model?

**Comment by btrotta** [comment]
@wyfSunflower Please open a new issue for this new question. Also, could you provide some more detail about what you're trying to do? I don't understand what you mean by "gbdt+lr model".
